Code/Legacy/CrySystem/SystemInit.cpp

```diff
@@ -892,16 +892,19 @@ void CSystem::InitLocalization()
         OpenLanguagePak(language);
     }
 
-    pCVar = m_env.pConsole != 0 ? m_env.pConsole->GetCVar("g_languageAudio") : 0;
-    if (pCVar)
+    if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
     {
-        if (strlen(pCVar->GetString()) == 0)
+        AZ::CVarFixedString languageAudio;
+        if (auto result = console->GetCvarValue("g_languageAudio", languageAudio); result == AZ::GetValueResult::Success)
         {
-            pCVar->Set(language);
-        }
-        else
-        {
-            language = pCVar->GetString();
+            if (languageAudio.size() == 0)
+            {
+                console->PerformCommand(AZStd::string::format("g_languageAudio %s", language.c_str()).c_str());
+            }
+            else
+            {
+                language.assign(languageAudio.data(), languageAudio.size());
+            }
```

Comment:
This logic doesn't really make any sense.
If the `g_lanugageAudio` CVar has a value, then set the localization `language` to it, but if it doesn't set it to the localization `langauge`

Reply:
Yeah the logic is similar to the original: `language` is a local variable, it *maybe* overwrites it with `g_languageAudio` before passing it to a function below.